appconfig.go

```diff
@@ -8,22 +8,24 @@ import (
 // Client contains the endpoint, ID and Secret for accessing
 // Azure Client service.  Create with New()
 type Client struct {
-	Endpoint string
-	ID       string
-	Secret   string
-	client   *http.Client
+	Endpoint   string
+	ID         string
+	Secret     string
+	client     *http.Client
+	cfg *ClientConfig,
 }
 
 // New creates a new Client instance from the
 // connection string provided in the Azure Portal.
-func New(connectionString string) *Client {
+func New(connectionString string, cfg *ClientConfig) *Client {
```

Comment:
Having a fluent `WithConfig(cfg ClientConfig) *Client` function that set the cfg would fit the pattern that Brian started in his code (see WithHttpClient). 

Once we have done that, we have the option of tweaking how ClientConfig works. In the `New(connectionString string)` function we can default `cfg` to a default instance of `ClientConfig{}` and the functions like `APIVersion()` can instead of looking for nil, can check for `c.version == ""` and then return the default. What do you think?